interfaces/cython/cantera/composite.py

```diff
@@ -637,22 +647,68 @@ def collect_data(self, cols=('extra','T','density','Y'), threshold=0,
 
         return np.hstack(data), labels
 
-    def write_csv(self, filename, cols=('extra','T','density','Y'),
+    def write_csv(self, file_name, cols=('extra','T','density','Y'),
                   *args, **kwargs):
         """
-        Write a CSV file named *filename* containing the data specified by
+        Write a CSV file named *file_name* containing the data specified by
         *cols*. The first row of the CSV file will contain column labels.
 
         Additional arguments are passed on to `collect_data`. This method works
         only with 1D SolutionArray objects.
         """
         data, labels = self.collect_data(cols, *args, **kwargs)
-        with open(filename, 'w') as outfile:
+        with open(file_name, 'w') as outfile:
             writer = _csv.writer(outfile)
             writer.writerow(labels)
             for row in data:
                 writer.writerow(row)
 
+    def to_pandas(self, cols=('extra','T','density','Y'),
+                  *args, **kwargs):
+        """
+        Returns the data specified by *cols* in a single pandas DataFrame.
+
+        Additional arguments are passed on to `collect_data`. This method works
+        only with 1D SolutionArray objects and requires a working pandas
+        installation.
+        """
+
+        # local import avoids explicit dependence of cantera on pandas
+        import pandas as pd
+
+        data, labels = self.collect_data(cols, *args, **kwargs)
+        return pd.DataFrame(data=data, columns=labels)
+
+    def write_hdf(self, file_name, cols=('extra','T','density','Y'),
+                  key='df', mode='a', append=False, complevel=5, *args, **kwargs):
```

Comment:
How did you pick this `complevel`? Why use the `a` file mode and not `w`? Can you document what happens to the `*args` and `**kwargs` arguments? Can you add spaces after the commas in the default argument for `cols`?

Reply:
OK. Spaces aren't a problem: I am refraining from autopep8'ing the file, but will add the spaces wherever I can find them
* Regarding default values: I was trying to keep things in line with the pandas defaults. It may be cleaner to set them all to None, and construct a kwarg dictionary so it always defaults to pandas, no matter what they do. As an aside, the mid-range `complevel` was a decent compromise between compression and speed.
* `*args` and `**kwargs` are documented with `collect_data`, and was following the approach taken for `write_csv` ...

Reply:
PS: the append `a` mode is a good choice as you can open existing files and add new data (with a different key) at a later point without overwriting what's already there.

Reply:
re the spaces: This is a difference that @speth and I have for function arguments, I prefer to be more strict about following PEP8 where he's a little more flexible. I would say that code that you're adding/modifying, add them where it makes sense, but I wouldn't bother in other code. Where it makes sense = some combination of avoiding weird line breaks and following PEP8

re defaults: I think it would be better to default to the Pandas defaults with `None` values.

re `*args`: I meant document as being passed through to `collect_data()`, if that's what's happening to them (they could also be passed through to `pandas.write_hdf()`).

Reply:
I know it's not strictly PEP8-compliant, but my rationale for this specific case (literal tuple as an argument) is that I think it makes it easier to see the argument and its default value as a single unit (which I think is the point of the PEP8 recommendation of not having spaces around the `=` here), while adding the spaces inside the tuple makes it easier to misread elements of the tuple as being individual arguments to the function.

Reply:
@speth and @bryanwweber ... I see both points, but they're conflicting. I do believe there is enough visual clue from `cols=(`, so I'm leaning towards PEP8. Will run with this until I hear otherwise.